packages/evm/src/evm.ts

```diff
@@ -221,7 +221,16 @@ export class EVM implements EVMInterface {
     this._bn254 = opts.bn254!
 
     this._emit = async (topic: string, data: any): Promise<void> => {
-      return new Promise((resolve) => this.events.emit(topic as keyof EVMEvents, data, resolve))
+      const listeners = this.events.listeners(topic as keyof EVMEvent)
+      for (const listener of listeners) {
+        if (listener.length === 2) {
+          await new Promise<void>((resolve) => {
+            listener(data, resolve)
+          })
+        } else {
+          listener(data)
```

Comment:
This is the async logic handler now, right? This is not part of `eventemitter3`?

Reply:
Correct, if the listener has 2 arguments, we assume it's looking for the event object and then the resolve function (and therefore is async).  Otherwise, we just call the listener.